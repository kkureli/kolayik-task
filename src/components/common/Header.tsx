import React, {FC} from 'react';
import {Image, SafeAreaView, StyleSheet, TouchableOpacity} from 'react-native';
import {hitSlop} from '../../theme/constants';
import Typography from './Typography';

interface HeaderProps {
  canGoBack?: boolean;
  isMore?: boolean;
  title?: string;
  onBackPress?: Function;
  onMorePress?: Function;
}

const Header: FC<HeaderProps> = ({
  canGoBack,
  isMore,
  title,
  onBackPress,
  onMorePress
}) => {
  return (
    <SafeAreaView style={styles.container}>
      {canGoBack && (
        <TouchableOpacity
          hitSlop={hitSlop}
          onPress={() =>
            onBackPress ? onBackPress() : console.log('Back pressed')
          }>
          <Image
            resizeMode="center"
            style={styles.icon}
            source={require('../../assets/icons/headerIcons/backIcon.png')}
          />
        </TouchableOpacity>
      )}
      {title && <Typography bold>{title}</Typography>}
      {isMore && (
        <TouchableOpacity
          hitSlop={hitSlop}
          onPress={() =>
            onMorePress ? onMorePress() : console.log('More pressed')
          }>
          <Image
            resizeMode="center"
            style={styles.icon}
            source={require('../../assets/icons/headerIcons/moreIcon.png')}
          />
        </TouchableOpacity>
      )}
    </SafeAreaView>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginHorizontal: 10
  },
  icon: {width: 24, height: 24}
});
