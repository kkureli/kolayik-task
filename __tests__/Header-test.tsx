import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/components/common/Header';

test('header renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
