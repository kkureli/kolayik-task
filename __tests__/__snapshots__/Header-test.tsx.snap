// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 10,
    }
  }
/>
`;
